pkg/storage/snapshot/source.go

```diff
@@ -407,6 +406,49 @@ func (s *vmSnapshotSource) Spec() (snapshotv1.SourceSpec, error) {
 	}, nil
 }
 
+func (s *vmSnapshotSource) handleVolumes(vmCpy *snapshotv1.VirtualMachine) error {
+	vmi, exists, err := s.controller.getVMI(s.vm)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return fmt.Errorf("can't evaluate volumes for online snapshot, vmi doesn't exist")
+	}
+
+	vmDiskMap := map[string]kubevirtv1.Disk{}
+	for _, disk := range s.vm.Spec.Template.Spec.Domain.Devices.Disks {
+		vmDiskMap[disk.Name] = disk
+	}
+
+	vmiVolNames := sets.New[string]()
+	for _, vol := range vmi.Spec.Volumes {
+		vmiVolNames.Insert(vol.Name)
+	}
+
+	vmiDiskNames := sets.New[string]()
+	for _, disk := range vmi.Spec.Domain.Devices.Disks {
+		vmiDiskNames.Insert(disk.Name)
+	}
+
+	vmCpy.Spec.Template.Spec.Volumes = []kubevirtv1.Volume{}
+	vmCpy.Spec.Template.Spec.Domain.Devices.Disks = []kubevirtv1.Disk{}
```

Comment:
Sorry to come back to this @ShellyKa13 and @Acedus but why cant this function simply set the following?

```golang
vmCpy.Spec.Template.Spec.Volumes = vmi.Spec.Volumes
vmCpy.Spec.Template.Spec.Domain.Devices.Disks = vmi.Spec.Domain.Devices.Disks
```

Reply:
good point, so can this function only look at hotpluggable volumes?